module3/lessons/Week2/IntroToMVC.md

```diff
@@ -0,0 +1,307 @@
+1. Register the context in Program.cs using the database connection string and snake_case naming convention
+
+To run this application, click the outlined "play" button to "Start Without Debugging". You will probably see the following confirmations
+
+![Creating a Web App Confirmation Messages](/assets/images/module3/Week2/MVC_2_Create_Web_App_Confirmations.png)
+
+When you click start, Visual Studio runs the app and opens it in the default browser. You should see a page that says "Welcome Learn about building Web apps with ASP.NET Core."
+
+The address bar shows `localhost:<port#>`. Localhost means the application is running on your local computer. When Visual Studio creates a web project, a random port is used for the web server.
+
+## 2. Add a Controller
+
+<aside class="instructor-notes">
+    <p><strong>Instructor Note</strong><br>I encourage referencing back to the sides from the start of class to refresh what a controller is before diving into this section. Same thing for the View and Model.
+    </p>
+</aside>
+
+We're going to use the scaffolding tool in Visual Studio to create our controller. Follow the instructions in the following image, except call your controller `MoviesController.cs` instead of `HelloWorldController.cs`
+
+![Creating a Controller](/assets/images/module3/Week2/MVC_3_Create_Controller.png)
+
+Your generated controller should look like this:
+```C#
+using Microsoft.AspNetCore.Mvc;
+
+namespace MvcMovie.Controllers
+{
+    public class MoviesController : Controller
+    {
+        //GET: /Movies/
+        public IActionResult Index()
+        {
+            return View();
+        }
+    }
+}
+```
+
+Every public method in a controller is callable as an HTTP endpoint.
+
+An HTTP endpoint:
+* Is a targetable URL in the web application, such as `https://localhost:<port#>/Movies`.
+
+Combines:
+* The protocol used: HTTPS.
+* The network location of the web server, including the port: localhost:5001.
+* The path: Movies.
+
+In your browser, navigate to `https://localhost:<port#>/Movies`.
+
+You should get an Error that looks like the following:
+![Error message before creating view](/assets/images/module3/Week2/MVC_View_Error_Message.png)
+
+`InvalidOperationException: The view 'Index' was not found. The following locations were searched: /Views/Movies/Index.cshtml /Views/Shared/Index.cshtml`
+
+We know from the Request and Response cycle lesson that our application needs to return something in response to the HTTPS request. We are trying to return `return View();`, but we haven't yet created that view!
+
+## 3. Add a View
+
+View templates are created using Razor.
+
+Razor-based view templates:
+* Have a .cshtml file extension.
+* Provide an elegant way to create HTML output with C#.
+
+We will use Visual Studio to create a new view. Let's all follow these steps, but instead of calling our folder `HelloWorld` we will call it `Movies`.
+
+![Adding a View](/assets/images/module3/Week2/MVC_5_Create_View.png)

```

Comment:
After adding the folder, when I right- clicked the folder, I had two different options that seem appropriate:
- `Add > View` or
- `Add > Razor Component`

Also, new screenshot (maybe?)

Reply:
Oh interesting, maybe that screenshot was from before the Razor stuff came out. Will look more into this.